lib/Raven/Client.php

```diff
@@ -112,13 +113,26 @@ class Client
     protected $config;
 
     /**
+     * @var HttpAsyncClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var MessageFactory The message factory
+     */
+    private $messageFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration   $config     The client configuration
+     * @param HttpAsyncClient $httpClient The HTTP client
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, HttpAsyncClient $httpClient, MessageFactory $messageFactory)
```

Comment:
> it's not useful

It is VERY useful. Also it promote us capability to extend code without breaking back compatibility.